### PR TITLE
fix error handling

### PR DESCRIPTION
**Pull request by pflanze**
This fixes error reporting for some kinds of invalid queries.

We ran into this while working on the boolean columns.

Does this warrant tests (I didn't really save the queries we made, sadly, but may be able to reconstruct them)?

### CODE REVIEW

**Comment by pflanze** [comment]
Sorry, will check the failing e2e tests later. I only ran the integration tests, need to include e2e in my script.

File: src/silo/query_engine/filter_expressions/string_equals.cpp [hunk 1]
```diff
@@ -46,21 +53,17 @@ std::unique_ptr<silo::query_engine::operators::Operator> StringEquals::compile(
...
-      if (embedded_string.has_value()) {
-         return std::make_unique<operators::Selection>(
-            std::make_unique<operators::CompareToValueSelection<common::SiloString>>(
-               string_column.getValues(), operators::Comparator::EQUALS, embedded_string.value()
-            ),
-            database_partition.sequence_count
-         );
-      }
-      return std::make_unique<operators::Empty>(database_partition.sequence_count);
+   // database_partition.columns.string_columns.contains(column)
```
Comment (fengelniederhammer) [inline src/silo/query_engine/filter_expressions/string_equals.cpp:56]: Let's delete this?

**Review by fengelniederhammer — CHANGES_REQUESTED** [review]
I think the `DateBetween` and `IntBetween` and filters could use the same check. 

```
{
  "action": {
    "limit": 100,
    "type": "Aggregated"
  },
  "filterExpression": {
    "children": [
      {
        "column": "division",
        "from": "Aargau",
        "to": "Aargau",
        "type": "DateBetween"
      }
    ],
    "type": "And"
  }
}
``` 
fails with 
```
{
  "error": "Internal Server Error",
  "message": "map::at"
}
``` 

Otherwise, this is a nice improvement :+1: 

In the long run, it would be nice if we could also test error messages in the unit tests. IIRC, that's currently not possible.

File: src/silo/query_engine/filter_expressions/string_equals.cpp (hunk 1 above)
Comment (Taepper) [reply src/silo/query_engine/filter_expressions/string_equals.cpp:56]: I would like to turn this into:
`database_partition.columns.string_columns.contains(column) `

**Comment by pflanze** [comment]
I've added @fengelniederhammer 's test example to the commit message, so that it's easy to find if in the future someone wants to write actual tests. I think it's ready to re-review.

**Comment by pflanze** [comment]
I've now also added single quotes around the column names in the error messages.

**Review by Taepper — APPROVED** [review]

**Pull request merged**